tests/pipeline/test_measuring.py

```diff
@@ -16,85 +16,80 @@ def test_measuring(measurer, decam_cutouts, decam_default_calibrators):
-    decam_cutouts[8].sub_flags[sz[0] // 2 + 2, sz[1] // 2 + 1] = 1  # bad pixel
-    decam_cutouts[8].sub_flags[sz[0] // 2 - 2, sz[1] // 2 + 1] = 4  # saturated should be ignored!
+    decam_cutouts.co_dict[f"source_index_8"]["sub_data"] = np.zeros_like(decam_cutouts.co_dict[f"source_index_0"]["sub_data"])
+    decam_cutouts.co_dict[f"source_index_8"]["sub_data"][sz[0] // 2, sz[1] // 2] = 100.0
+    decam_cutouts.co_dict[f"source_index_8"]["sub_flags"][sz[0] // 2 + 2, sz[1] // 2 + 1] = 1  # bad pixel
+    decam_cutouts.co_dict[f"source_index_8"]["sub_flags"][sz[0] // 2 - 2, sz[1] // 2 + 1] = 4  # saturated should be ignored!
 
     # delta function with offset that makes it far from the bad pixel
-    decam_cutouts[9].sub_data = np.zeros_like(decam_cutouts[0].sub_data)
-    decam_cutouts[9].sub_data[sz[0] // 2 + 3, sz[1] // 2 + 3] = 100.0
-    decam_cutouts[9].sub_flags[sz[0] // 2 - 2, sz[1] // 2 - 2] = 1  # bad pixel
+    decam_cutouts.co_dict[f"source_index_9"]["sub_data"] = np.zeros_like(decam_cutouts.co_dict[f"source_index_0"]["sub_data"])
+    decam_cutouts.co_dict[f"source_index_9"]["sub_data"][sz[0] // 2 + 3, sz[1] // 2 + 3] = 100.0
+    decam_cutouts.co_dict[f"source_index_9"]["sub_flags"][sz[0] // 2 - 2, sz[1] // 2 - 2] = 1  # bad pixel
 
     # gaussian that is too wide
-    decam_cutouts[10].sub_data = make_gaussian(imsize=sz[0], sigma_x=fwhm / 2.355 * 2, norm=1) * 1000
-    decam_cutouts[10].sub_data += np.random.normal(0, 1, size=sz)
+    decam_cutouts.co_dict[f"source_index_10"]["sub_data"] = make_gaussian(imsize=sz[0], sigma_x=fwhm / 2.355 * 2, norm=1) * 1000
+    decam_cutouts.co_dict[f"source_index_10"]["sub_data"] += np.random.normal(0, 1, size=sz)
 
     # streak
-    decam_cutouts[11].sub_data = make_gaussian(imsize=sz[0], sigma_x=fwhm / 2.355, sigma_y=20, rotation=25, norm=1)
-    decam_cutouts[11].sub_data *= 1000
-    decam_cutouts[11].sub_data += np.random.normal(0, 1, size=sz)
-
-    # a regular cutout but we'll put some bad flag on the cutout
-    decam_cutouts[12].badness = 'cosmic ray'
-
-    # a regular cutout with a bad flag that we are ignoring:
-    decam_cutouts[13].badness = 'satellite'
+    decam_cutouts.co_dict[f"source_index_11"]["sub_data"] = make_gaussian(imsize=sz[0], sigma_x=fwhm / 2.355, sigma_y=20, rotation=25, norm=1)
+    decam_cutouts.co_dict[f"source_index_11"]["sub_data"] *= 1000
+    decam_cutouts.co_dict[f"source_index_11"]["sub_data"] += np.random.normal(0, 1, size=sz)
 
     # run the measurer
     ds = measurer.run(decam_cutouts)
 
-    assert len(ds.all_measurements) == len(ds.cutouts)
+    assert len(ds.all_measurements) == len(ds.cutouts.co_dict)
 
     # verify all scores have been assigned
     for score in measurer.pars.analytical_cuts:
         assert score in ds.measurements[0].disqualifier_scores
 
-    m = ds.all_measurements[0]  # delta function
+    m = [m for m in ds.all_measurements if m.index_in_sources == 0][0]  # delta function
```

Comment:
I'm surprised that the order of `all_measurements` is not the same as the `index_in_sources`. If you'd do `all_measurements[i]` wouldn't you always get `index_in_sources==i`?

Reply:
I did not get the same order, the test ended up failing a few indices in (at the gaussian), until I made this modification. I didn't look too deeply into why the order was not the same however. I will look a little bit more before I merge since it seems curious.